rpc/erigon.go

```diff
@@ -768,3 +770,339 @@ type Eth1InternalTransactionWithPosition struct {
+	Hash          string                    `json:"hash"`
+	ParentHash    string                    `json:"parentHash"`
+	UncleHash     string                    `json:"uncleHash"`
+	Coinbase      string                    `json:"coinbase"`
+	Root          string                    `json:"stateRoot"`
+	TxHash        string                    `json:"transactionsHash"`
+	ReceiptHash   string                    `json:"receiptsHash"`
+	Difficulty    string                    `json:"difficulty"`
+	Number        string                    `json:"number"`
+	GasLimit      string                    `json:"gasLimit"`
+	GasUsed       string                    `json:"gasUsed"`
+	Time          string                    `json:"timestamp"`
+	Extra         string                    `json:"extraData"`
+	MixDigest     string                    `json:"mixHash"`
+	Bloom         string                    `json:"logsBloom"`
+	Uncles        []*geth_types.Block       `json:"uncles"`
+	Transactions  []*geth_types.Transaction `json:"transactions"`
+	Withdrawals   []*geth_types.Withdrawal  `json:"withdrawals"`
+	BlobGasUsed   *string                   `json:"blobGasUsed"`
+	ExcessBlobGas *string                   `json:"excessBlobGas"`
+	BaseFee       string                    `json:"baseFee"`
+}
+
+func (b *BlockResponse) UnmarshalJSON(data []byte) error {
+	type Alias BlockResponse
+	tempData := &struct {
+		BlobGasUsed   json.RawMessage `json:"blobGasUsed"`
+		ExcessBlobGas json.RawMessage `json:"excessBlobGas"`
+		Uncles        json.RawMessage `json:"uncles"`
+		*Alias
+	}{
+		Alias: (*Alias)(b),
+	}
+
+	if err := json.Unmarshal(data, &tempData); err != nil {
+		return err
+	}
+
+	if string(tempData.BlobGasUsed) != "" && string(tempData.BlobGasUsed) != "null" {
+		*b.BlobGasUsed = string(tempData.BlobGasUsed)
+	} else {
+		b.BlobGasUsed = nil
+	}
+
+	if string(tempData.ExcessBlobGas) != "" && string(tempData.ExcessBlobGas) != "null" {
+		*b.ExcessBlobGas = string(tempData.ExcessBlobGas)
+	} else {
+		b.ExcessBlobGas = nil
+	}
+
+	if string(tempData.Uncles) == "null" || string(tempData.Uncles) == "" {
+		b.Uncles = []*geth_types.Block{}
+	} else if tempData.Uncles[0] == '[' {
+		// unmarshal as an array of type geth_types.Block
+		if err := json.Unmarshal(tempData.Uncles, &b.Uncles); err != nil {
+			// if it failed then unmarshal as an array of strings
+			var uncleHashes []string
+			if err := json.Unmarshal(tempData.Uncles, &uncleHashes); err != nil {
+				return fmt.Errorf("expected an array for uncles, but got: %s", tempData.Uncles)
+			}
+			// convert string hashes to empty Block refs
+			for i := range uncleHashes {
+				b.Uncles[i] = &geth_types.Block{}
+			}
+		}
+	} else {
+		return fmt.Errorf("expected an array for uncles, but got: %s", tempData.Uncles)
+	}
+
+	return nil
+}
+
+func (client *ErigonClient) GetBlocksByBatch(blocksChan chan *types.Eth1Block, erigonClient *ErigonClient) ([]*types.Eth1Block, *types.GetBlockTimings, error) {
```

Comment:
wrong signature, no need to receive `erigonClient *ErigonClient`

Reply:
The Erigon client was always nil during my tests, which is why I passed it here